exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/MetadataContext.java

```diff
@@ -41,6 +41,10 @@
 
   private PruneStatus pruneStatus = PruneStatus.NOT_STARTED;
 
+  // False values of these flags allow to avoid double reading of corrupted or unsupported metadata files
+  public boolean isMetaCacheFileCorrect = true;
+  public boolean isMetaDirsCacheFileCorrect = true;
```

Comment:
Nice!

How are these states cleared? Do we keep track of the file timestamp, and try again if the file is replaced/updated? Or, is there some kind of manual reset? Is the state only per query?